.github/workflows/ci.yaml

```diff
@@ -8,177 +8,303 @@ on:
-        uses: actions/upload-artifact@v2
+
+      - name: Install node.js v12
+        uses: actions/setup-node@v2
         with:
-          name: release-packages
-          path: ./release-packages
+          node-version: "12"
+
+      - name: Install nfpm
+        run: |
+          curl -sfL https://install.goreleaser.com/github.com/goreleaser/nfpm.sh | sh -s -- -b ~/.local/bin v2.3.1
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
 
-  linux-arm64:
-    needs: release
-    runs-on: ubuntu-arm64-latest
-    steps:
-      - uses: actions/checkout@v2
       - name: Download npm package
         uses: actions/download-artifact@v2
         with:
           name: npm-package
-          path: ./release-npm-package
-      - name: Run ./ci/steps/release-packages.sh
-        uses: ./ci/images/centos7
-        with:
-          args: ./ci/steps/release-packages.sh
+
+      - name: Decompress npm package
+        run: tar -xzf package.tar.gz
+
+      - name: Build standalone release
+        run: yarn release:standalone
+
+      - name: Sanity test standalone release
+        run: yarn test:standalone-release
+
+      - name: Build packages with nfpm
+        run: yarn package
+
       - name: Upload release artifacts
         uses: actions/upload-artifact@v2
         with:
           name: release-packages
           path: ./release-packages
-      - name: Remove docker images
-        run: docker system prune -af
-
-  macos-amd64:
-    needs: release
-    runs-on: macos-latest
-    # This job requires secrets, so can only run on the default branch
-    if: github.ref == 'refs/heads/main'
+
+  package-arm64:
+    name: Linux ARM64 cross-compile build
+    needs: build
+    runs-on: ubuntu-16.04
+    env:
+      AR: aarch64-linux-gnu-ar
+      CC: aarch64-linux-gnu-gcc
+      CXX: aarch64-linux-gnu-g++
+      LINK: aarch64-linux-gnu-g++
+      NPM_CONFIG_ARCH: arm64
+
     steps:
       - uses: actions/checkout@v2
+
+      - name: Install node.js v12
+        uses: actions/setup-node@v2
+        with:
+          node-version: "12"
+
+      - name: Install nfpm
+        run: |
+          curl -sfL https://install.goreleaser.com/github.com/goreleaser/nfpm.sh | sh -s -- -b ~/.local/bin v2.3.1
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: Install cross-compiler
+        run: sudo apt install g++-aarch64-linux-gnu
+
       - name: Download npm package
         uses: actions/download-artifact@v2
         with:
           name: npm-package
-          path: ./release-npm-package
-      - run: ./ci/steps/release-packages.sh
-        env:
-          # Otherwise we get rate limited when fetching the ripgrep binary.
-          # For whatever reason only MacOS needs it.
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Decompress npm package
+        run: tar -xzf package.tar.gz
+
+      - name: Build standalone release
+        run: yarn release:standalone
+
+      - name: Replace node with arm64 equivalent
```

Comment:
How come we do this? (might be helpful to add a comment above)

Reply:
It's all the magic above - I set `AR`/`CC`/`CXX`/`LINK` to their `aarch64` cross compile variants in the job's `env`, and then yarn would use those to build; however qemu-user-static doesn't correctly run node aarch64 on 16.04 (tested locally), so I just sub in the "native"/x86-64 node with the arm64 node before we package.

Reply:
Alrighty then! Might be worth adding a comment here just summarizing that.

Reply:
Thoughts on adding a comment here?

Reply:
I thought that it made the most sense to explain the situation above the whole workflow, before I set the env vars etc., so I put it here - https://github.com/cdr/code-server/pull/2966/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR181-R187 (you might need to click `Load diff` for the lines to get highlighted)

Reply:
Ah, I missed that! LGTM!